packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -128,8 +127,6 @@ export class ProxyContext implements EditableTreeContext {
         }
         assert(this.withCursors.size === 0, 0x3c1 /* free should remove all cursors */);
         assert(this.withAnchors.size === 0, 0x3c2 /* free should remove all anchors */);
-        this.forest.removeDependent(this.observer);
```

Comment:
if we don't remove the dependent where where does it get removed? It seems like this introduces a memory leak (every context you make on a forest leaks the observer, even if you free the context).

Reply:
@CraigMacomber Upd. Sorry for a confusion, but actually that line must be removed. I have forgotten after past three weeks. The reason is provided in the test https://github.com/microsoft/FluidFramework/blob/f365c3e99a5e8ce3ca8d78609225d56b252be88a/packages/dds/tree/src/test/feature-libraries/editable-tree/editableTreeContext.spec.ts#L66
We have to manually free the context after every "frame" to avoid anchor leaks. If we cleanup the observer, there is currently no proper way to restore it, since the context is a "static" part of a SharedTree (same way the observer is a "static" part of the context). So, we literally kill the context if we do so.
Currently it's all very coupled there (the SharedTree, the Forest, the context, etc.), we might want to decouple that if somebody would need to have multiple contexts for the same local Forest.